connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/outbound/jobhandling/SpringConnectorJobHandler.java

```diff
@@ -76,14 +114,111 @@ public void handle(JobClient client, ActivatedJob job) {
 
   private void executeJob(JobClient client, ActivatedJob job) {
     try {
-      super.handle(client, job);
+      internalHandle(client, job);
     } catch (Exception e) {
       connectorsOutboundMetrics.increaseFailure(job);
       LOGGER.warn("Failed to handle job: {} of type: {}", job.getKey(), job.getType());
     }
   }
 
-  @Override
+  public void internalHandle(final JobClient client, final ActivatedJob job) {
+    LOGGER.info(
+        "Received job: {} of type: {} for tenant: {}",
+        job.getKey(),
+        job.getType(),
+        job.getTenantId());
+    ConnectorResult result = getConnectorResult(job);
+    processFinalResult(client, job, result);
+  }
+
+  private ConnectorResult getConnectorResult(ActivatedJob job) {
+    Duration retryBackoff = null;
+    try {
+      retryBackoff = getBackoffDuration(job);
+      var context =
+          new JobHandlerContext(
+              job, getSecretProvider(), validationProvider, documentFactory, objectMapper);
```

Comment:
WDYT of moving this into a separate class and use it like this: `JobHandlerContextFactory.createHandler(job)`? 

This would reduce the amount of dependencies of this job handler implementation and make it easier to read.

It could be configured for example in the `OutboundConnectorRuntimeConfiguration`?